zerver/views/messages.py

```diff
@@ -325,7 +325,8 @@ def by_sender(self, query: Query, operand: Union[str, int], maybe_negate: Condit
             if isinstance(operand, str):
                 sender = get_user_including_cross_realm(operand, self.user_realm)
             else:
-                sender = get_user_by_id_in_realm_including_cross_realm(operand, self.user_realm)
+                sender = access_user_by_id(self.user_profile, operand, allow_bots=True,
+                                           allow_cross_realm=True, read_only=True)
         except UserProfile.DoesNotExist:
             raise BadNarrowOperator('unknown user ' + str(operand))
```

Comment:
@timabbott  access_user_by_id checks for UserProfile.DoesNotExist exception internally and raises JsonableError whereas get_user_by_id_in_realm_including_cross_realm as well as get_user_including_cross_realm do not check for UserProfile.DoesNotExist exception internally. In many places, where they (get_user functions) have been used , they are called inside try..exept and UserProfile.DoesNotExist exeptions are handled differently that raising JsonableError.

So, to replace them with access_user_by_id or access_user_by_email , should these (access_user functions) be refactored to expect themselves getting called inside try..except by removing the try..except clause inside them ?

Reply:
I hadn't been imagining we'd be applying this change inside `zerver/views/messages.py` yet and think we should leave that for after we've converted `users.py`.  But I'm guessing the way we'll do that is to catch the `JsonableError`s here and convert the exception to a `BadNarrowOperator`.